File: mobile/lib/domain/repository/account_repository.dart
import '../model/account.dart';

abstract class AccountRepository {
  Future<void> create(Account account);
  Future<Account?> findById(String id);
  Future<List<Account>> listByFamily(String familyId, {bool includeArchived = false});
  Future<void> update(Account account);
  Future<void> softDelete(String id);
  Stream<List<Account>> watchByFamily(String familyId);
}

File: backend/src/main/kotlin/mymoney/domain/repository/AccountRepository.kt
package mymoney.domain.repository

import mymoney.domain.model.Account
import java.util.UUID

interface AccountRepository {
    suspend fun create(account: Account): Account
    suspend fun findById(id: UUID): Account?
    suspend fun listByFamily(familyId: UUID, includeArchived: Boolean = false): List<Account>
    suspend fun update(account: Account): Account
    suspend fun softDelete(id: UUID)
}
